Common/Field.cs

```diff
@@ -57,6 +57,38 @@ public static partial class Field
         }
 
         /// <summary>
+        /// Defines an average price that is equal to (O + H + L + C) / 4
+        /// </summary>
+        public static Func<BaseData, decimal> Average
+        {
+            get { return TradeBarPropertyOrValue(x => (x.Open + x.High + x.Low + x.Close) / 4m); }
+        }
+
+        /// <summary>
+        /// Defines an average price that is equal to (H + L) / 2
+        /// </summary>
+        public static Func<BaseData, decimal> Median
+        {
+            get { return TradeBarPropertyOrValue(x => (x.High + x.Low) / 2m); }
+        }
+
+        /// <summary>
+        /// Defines an average price that is equal to (H + L + C) / 3
+        /// </summary>
+        public static Func<BaseData, decimal> Typical
+        {
+            get { return TradeBarPropertyOrValue(x => (x.High + x.Low + x.Close) / 3m); }
+        }
+
+        /// <summary>
+        /// Defines an average price that is equal to (H + L + 2*C) / 4
+        /// </summary>
+        public static Func<BaseData, decimal> Weighted
```

Comment:
Sounds good to me! Great stuff. I'm going to fix the travis build, looks like a test is failing and then merge this in